content/en/blog/2023/waypoint-proxy-made-simple/index.md

```diff
@@ -0,0 +1,221 @@
+
+Review the listeners for `reviews` waypoint proxy:
+
+{{< text bash >}}
+$ istioctl proxy-config listener deploy/bookinfo-reviews-istio-waypoint --waypoint
+LISTENER              CHAIN                                                 MATCH                                         DESTINATION
+envoy://connect_originate                                                       ALL                                           Cluster: connect_originate
+envoy://main_internal inbound-vip|9080||reviews.default.svc.cluster.local-http  ip=10.96.104.108 -> port=9080                 Inline Route: /*
+envoy://main_internal direct-tcp                                            ip=10.244.2.14 -> ANY                         Cluster: encap
+envoy://main_internal direct-tcp                                            ip=10.244.1.6 -> ANY                          Cluster: encap
+envoy://main_internal direct-tcp                                            ip=10.244.2.11 -> ANY                         Cluster: encap
+envoy://main_internal direct-http                                           ip=10.244.2.11 -> application-protocol='h2c'  Cluster: encap
+envoy://main_internal direct-http                                           ip=10.244.2.11 -> application-protocol='http/1.1' Cluster: encap
+envoy://main_internal direct-http                                           ip=10.244.2.14 -> application-protocol='http/1.1' Cluster: encap
+envoy://main_internal direct-http                                           ip=10.244.2.14 -> application-protocol='h2c'  Cluster: encap
+envoy://main_internal direct-http                                           ip=10.244.1.6 -> application-protocol='h2c'   Cluster: encap
+envoy://main_internal direct-http                                           ip=10.244.1.6 -> application-protocol='http/1.1'  Cluster: encap
+envoy://connect_terminate default                                               ALL                                           Inline Route:
+{{< /text >}}
+
+For requests arriving on port `15008`, which by default is Istio’s inbound HBONE port, the waypoint proxy terminates the HBONE connection and forwards the request to the `main_internal` to enforce any workload policies such as AuthorizationPolicy. If you are not familiar with the [internal listener](https://www.envoyproxy.io/docs/envoy/latest/configuration/other_features/internal_listener), it is an Envoy listener that accepts user space connections without using the system network API. The `--waypoint` flag added to the command earlier shows the details of the `main_internal` listeners, their filter chains, chain matches, and destinations.
```

Comment:
I have reviewed this section a few times and I find myself getting a bit lost trying to make the connection between the inbound request on 15008 (HBONE) and the `inbound-vip|9080||reviews.default.svc.cluster.local-http` internal listener. From my understanding, a request `(dst: $REVIEWS_SVC_IP:$HBONE_PORT)` arrives at the waypoint's `connect_terminate` listener. This listener terminates the HBONE connection and routes the request, now using `$REVIEWS_SVC_PORT` as the dest port, to the `main_internal` cluster. The `main_internal` cluster is configured to route the request to the `main_internal` listener. The `main_internal` listener routes the request to the reviews v1/v2 service subsets based on the configured weight. I'm not sure if we want to provide this level of detail but I feel like we're missing a few pieces of information to help connect the dots.

Reply:
I think you are correct, it is not my intention to go that level of details tho.  Main point is to show the configurations (listeners/clusters/routes etc) are trimmed for waypoints automatically.